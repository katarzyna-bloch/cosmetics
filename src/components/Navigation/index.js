import styled from 'styled-components'

const Nav = styled.nav`
  top: 0;
  left: 0;
  position: sticky;
  background-color: lightgray;
  padding: 20px;
`

const Link = styled.a`
padding: 30px;
font-weight: 600;
text-decoration: none;
color: black;
`

const Search = styled.input``

const Navigation = () => (     
  <Nav>
    <Link as="a" href="/">Oczyszczanie</Link>
    <Link as="a" href="/">Demakija≈º</Link>
    <Link as="a" href="/">Pielegnacja</Link>
    <Search type='text' placeholder='szukaj'/>
  </Nav>
)

export default Navigation
