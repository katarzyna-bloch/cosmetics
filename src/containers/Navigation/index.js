import React, { useEffect } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import Nav from '../../components/Navigation'
import { requestCategories } from '../../redux/actions'

export const Navigation = ({ fetchCategories, categories }) => {
  useEffect(() => {
    fetchCategories()
  }, [])

  if (!categories) {
    return null
  }

  return (
    <Nav categories={categories} />
  )
}

const mapStateToProps = ({ categories }) => ({
  categories,
})

const mapDispatchToProps = (dispatch) => ({
  fetchCategories: () => dispatch(requestCategories()),
})

Navigation.propTypes = {
  fetchCategories: PropTypes.func.isRequired,
  categories: PropTypes.array,
}

export default connect(mapStateToProps, mapDispatchToProps)(Navigation)
