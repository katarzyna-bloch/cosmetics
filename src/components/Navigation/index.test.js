import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Navigation from '.'

it('should render correctly', () => {
  let wrapped = shallow(<Navigation />)
  expect(toJson(wrapped)).toMatchSnapshot()
})