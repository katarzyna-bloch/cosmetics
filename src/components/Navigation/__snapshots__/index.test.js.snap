// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<styled.nav>
  <Styled(Link)
    to="/home-spa"
  >
    Domowe SPA
  </Styled(Link)>
  <Styled(Link)
    to="/cleaning"
  >
    Oczyszczanie
  </Styled(Link)>
  <Styled(Link)
    to="/care"
  >
    PielÄ™gnacja
  </Styled(Link)>
  <styled.input
    placeholder="szukaj"
    type="text"
  />
</styled.nav>
`;
